pkg/bbgo/session.go

```diff
@@ -207,6 +213,8 @@ func NewExchangeSession(name string, exchange types.Exchange) *ExchangeSession {
 		Session: session,
 	}
 
+	session.marginInfoUpdater = NewMarginInfoUpdaterFromSession(session)
```

Comment:
add the margin flag check here:

if session.Margin {

}

make it clearer that we have a flag checking to wrap the component, don't hide the condition in the constructor.

we always assume a constructor always returns an object.